### PR TITLE
Allow the recommenders to use extra_data

### PR DESCRIPTION
**Pull request by Dexterp37**
This additionally enables the `LocaleRecommender` to use the 'locale' coming from the `extra_data`, if it is available.

### CODE REVIEW

File: taar/recommenders/locale_recommender.py [hunk 1]
```diff
@@ -25,11 +25,14 @@ def __init__(self):
...
-    def can_recommend(self, client_data):
+    def can_recommend(self, client_data, extra_data={}):
         # We can't recommend if we don't have our data files.
         if self.top_addons_per_locale is None:
             return False
-        client_locale = client_data.get('locale', None)
+
+        # If we have data coming from other sources, we can use that for
+        # recommending.
+        client_locale = client_data.get('locale', None) or extra_data.get('locale', None)
```
Comment (mlopatka) [inline taar/recommenders/locale_recommender.py]: We must also check if the value contained in `extra_data.get('locale', None)` is found in `self.top_addons_per_locale`

File: taar/recommenders/locale_recommender.py [hunk 2]
```diff
@@ -41,6 +44,6 @@ def can_recommend(self, client_data):
 
         return True
 
-    def recommend(self, client_data, limit):
-        client_locale = client_data.get('locale')
+    def recommend(self, client_data, limit, extra_data={}):
+        client_locale = client_data.get('locale') or extra_data.get('locale', None)
```
Comment (mlopatka) [inline taar/recommenders/locale_recommender.py]: Please comment that we have chosen to give priority to the client data from telemetry/hbase over the data source coming from AMO

File: tests/test_localerecommender.py [hunk 3]
```diff
@@ -60,3 +60,19 @@ def test_recommender_str(mock_s3_json_downloader):
     # Tests that the string representation of the recommender is correct
     r = LocaleRecommender()
     assert str(r) == "LocaleRecommender"
+
+
+def test_recommender_extra_data(mock_s3_json_downloader):
+    # Test that the recommender uses locale data from the "extra"
+    # section if available.
+    r = LocaleRecommender()
+    recommendations = r.recommend({}, 10, extra_data={"locale": "en"})
```
Comment (mlopatka) [inline tests/test_localerecommender.py]: Please add test coverage for when the client data and the extra data provided different locale values.

File: taar/recommenders/locale_recommender.py (hunk 1 above)
Comment (Dexterp37) [reply taar/recommenders/locale_recommender.py]: That's done below, in the checks at lines 39-43

File: taar/recommenders/locale_recommender.py (hunk 2 above)
Comment (Dexterp37) [reply taar/recommenders/locale_recommender.py]: Done

File: tests/test_localerecommender.py (hunk 3 above)
Comment (Dexterp37) [reply tests/test_localerecommender.py]: Done

**Pull request merged**